sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/consumer/strategies/MessageListenerStrategy.java

```diff
@@ -61,6 +91,58 @@ public void init(ConsumerTopicMessageHistogram consumerTopicMessageHistogram) {
    */
   public abstract void processRecords(ConsumerRecords<K, V> records, KafkaConsumer<K, V> consumer);
 
+  /**
+   * Creates a context for the current {@link Thread} to handle a message with appropriate
+   * information in {@link ThreadLocal}s.
+   *
+   * <p>Example usage (notice the nested "try" to keep the context for error handling):
+   *
+   * <pre>
+   *   <code>try (var ignored = messageHandlerContextFor(consumerRecord)) {
+   *     try {
+   *       messageHandler.handle(consumerRecord);
+   *     } catch (Exception e) {
+   *       errorHandler.handleError(consumerRecord, e, consumer);
+   *     }
+   *   }
+   *   </code>
+   * </pre>
+   *
+   * <p>The context will be set up as follows:
+   *
+   * <ol>
+   *   <li>Sets the {@link MetadataContext} for the current thread to handle the given {@code
+   *       consumerContext} with the desired context. The context information is based on the {@link
+   *       ConsumerRecord#headers()} considering all configured metadata fields.
+   *       <p>This method should be used as try with resources around the message handler to
```

Comment:
nit: usually I see this notation in official documentations: "try-with-resources"

Reply:
fixed